### PR TITLE
Fix Copy with UseSymbolicLinksIfPossible = true

### PR DESCRIPTION
**Pull request by Forgind**
### Changes Made
Fix symlink creation

### Testing
I tried creating a symlink with this, and it succeeded, both in the unit test and in the wild. I also tweaked the unit test to verify that the reparse point is not 0 for the created file. I removed the early exit if the user doesn't have the expected permission.

### Notes
Once both this and #8149 are merged, we should modify the test in #8149 to use this to create a symlink on windows.

### CODE REVIEW

File: src/Tasks.UnitTests/Copy_Tests.cs [hunk 1]
```diff
@@ -2397,77 +2397,63 @@ public CopySymbolicLink_Tests(ITestOutputHelper testOutputHelper)
...
+                };
 
-                    bool success = t.Execute();
+                bool success = t.Execute();
 
-                    Assert.True(success); // "success"
-                    Assert.True(File.Exists(destFile)); // "destination exists"
+                Assert.True(success); // "success"
+                Assert.True(File.Exists(destFile)); // "destination exists"
+                Assert.True((File.GetAttributes(destFile) & FileAttributes.ReparsePoint) != 0);
```
Comment (rainersigwald) [inline src/Tasks.UnitTests/Copy_Tests.cs:2426]: I wish we also had tests for the fallback-to-copy behavior but we didn't have them before so oh well.

Comment (Forgind) [reply src/Tasks.UnitTests/Copy_Tests.cs:2426]: This seems reasonable, but it also seems like a hard test to write. Either you can create symlinks or you can't. You mentioned there's a limit to how many symlinks you can create for a given file; looking online, it looks like that number is 31. I can try to make 32 symlinks, but that sounds like an unwieldy and somewhat expensive test. Up to you.

File: src/Tasks.UnitTests/Copy_Tests.cs [hunk 2]
```diff
@@ -2397,77 +2397,71 @@ public CopySymbolicLink_Tests(ITestOutputHelper testOutputHelper)
         [Fact]
         public void CopyToDestinationFolderWithSymbolicLinkCheck()
         {
-            var isPrivileged = true;
-
-            if (NativeMethodsShared.IsWindows)
+            Version osVersion = Environment.OSVersion.Version;
+            if (NativeMethodsShared.IsWindows && osVersion.Major < 11 && (osVersion.Major < 10 || osVersion.Build < 14972))
```
Comment (Forgind) [inline src/Tasks.UnitTests/Copy_Tests.cs:2401]: delete this check; put check in other system configuration tests

File: src/Tasks.UnitTests/Copy_Tests.cs (hunk 1 above)
Comment (rainersigwald) [inline src/Tasks.UnitTests/Copy_Tests.cs:2426]:
```suggestion
                Assert.True((File.GetAttributes(destFile) & FileAttributes.ReparsePoint) != 0, "File was copied but is not a symlink");
```

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**